### PR TITLE
feat: option to set path destination to parent folder when cursor is on a closed folder while creating files

### PR DESCRIPTION
**Pull request by booperlv**
a more thorough description of the feature in #622 
Seems like this is the first instance of an option for the `fs.lua` module, so I decided to go with directly adding the variable there.
This was the only name I could think of that made sense, but I'm not sure if it makes _enough_ sense, so please do help me think of a better name if so.

Please do tell me if there's anything that I should change! Thanks :)

### CODE REVIEW

**Comment by CKolkey** [comment]
I've had something similar in a fork for a while:

```lua
  local add_into
  if node.entries ~= nil and node.open then
    add_into = utils.path_add_trailing(node.absolute_path)
  else
    add_into = node.absolute_path:sub(0, -(#node.name + 1))
  end
```

**Comment by booperlv** [comment]
> I've had something similar in a fork for a while:
> 
> ```lua
>   local add_into
>   if node.entries ~= nil and node.open then
>     add_into = utils.path_add_trailing(node.absolute_path)
>   else
>     add_into = node.absolute_path:sub(0, -(#node.name + 1))
>   end
> ```

This does seem like a more robust solution! I'll change it accordingly :)

File: lua/nvim-tree/fs.lua [hunk 1]
```diff
@@ -57,8 +57,13 @@ function M.create(node)
     }
   end
 
+  local node_is_open = (
+    vim.g.nvim_tree_create_in_closed_folder == 0
+    and node.open or false
```
Comment (kyazdani42) [inline lua/nvim-tree/fs.lua:62]: this feature is not working and breaks default behavior.
`local node_is_open = vim.g.nvim_tree_create_in_closed_folder == 1 or node.open` should be more appropriate if i understand the feature correctly.
With you condition, setting `0` or `1` for `nvim_tree_create_in_closed_folder` does not change anything and always creates the file outside of the folder.

**Review by kyazdani42 — CHANGES_REQUESTED** [review]

File: lua/nvim-tree/fs.lua (hunk 1 above)
Comment (booperlv) [reply lua/nvim-tree/fs.lua:62]: Apologies, sleep deprivation has gotten the better of me again. Pushing a fix in a few!

**Review by kyazdani42 — APPROVED** [review]

**Pull request merged**